src/oisst/State/State.cc

```diff
@@ -5,120 +5,366 @@
+    float sstData[lat][lon];
+    sstVar.getVar(sstData);
+
+    int idx = 0;
+    for (int i = 0; i < lat; i++)
+      for (int j = 0; j < lon; j++)
+        field_data(idx++) = sstData[i][j];
   }
 
 // ----------------------------------------------------------------------------
 
   void State::write(const eckit::Configuration & conf) const {
-    util::abor1_cpp("State::write() needs to be implemented.",
-                     __FILE__, __LINE__);
+    int lat, lon, time = 1;
+    std::string filename;
+
+    // get filename
+    if (!conf.get("filename", filename))
+      util::abor1_cpp("Get filename failed.", __FILE__, __LINE__);
+    else
+      std::cout << "State::write(), filename=" << filename << std::endl;
+
+    // create netCDF file
+    netCDF::NcFile file(filename.c_str(), netCDF::NcFile::replace);
+    if (file.isNull())
+      util::abor1_cpp("Create netCDF file failed.", __FILE__, __LINE__);
+
+    // define dims
+    lat = geom_->atlasFunctionSpace()->grid().ny();
+    lon =((atlas::RegularLonLatGrid)(geom_->atlasFunctionSpace()->grid())).nx();
+
+    // unlimited dim if without size parameter, then it'll be 0,
+    // what about the size?
+    netCDF::NcDim timeDim = file.addDim("time", 1);
+    netCDF::NcDim latDim  = file.addDim("lat" , lat);
+    netCDF::NcDim lonDim  = file.addDim("lon" , lon);
+    if (timeDim.isNull() || latDim.isNull() || lonDim.isNull())
+      util::abor1_cpp("Define dims failed.", __FILE__, __LINE__);
+
+    std::vector<netCDF::NcDim> dims;
+    dims.push_back(timeDim);
+    dims.push_back(latDim);
+    dims.push_back(lonDim);
+
+    // Lignag: define coordinate vars "lat" and "lon"
+
+    // Ligang: define units atts for coordinate vars
+
+    // define data vars
+    netCDF::NcVar sstVar = file.addVar(std::string("sst"), netCDF::ncFloat
+                                       , dims);
+
+    // Ligang: define units atts for data vars
+    sstVar.putAtt("units", "K");
+    sstVar.putAtt("_FillValue", netCDF::NcFloat(), -32768.);
+
+    // write data to the file
+    // Ligang: compile failed, rank should be 1, not 2, related to init?
+//  auto field_data = make_view<float, 2>(atlasFieldSet_->field(0));
+    auto field_data = make_view<float, 1>(atlasFieldSet_->field(0));
+
+    float sstData[time][lat][lon];
+    int idx = 0;
+    for (int i = 0; i < lat; i++)
+      for (int j = 0; j < lon; j++)
+        sstData[0][i][j] = field_data(idx++);
+
+    sstVar.putVar(sstData);
+
+    oops::Log::info() << "Successfully write data to file!" << std::endl;
   }
 
-// ----------------------------------------------------------------------------
-
-  boost::shared_ptr<const Geometry> State::geometry() const {return geom_;}
-
 // ----------------------------------------------------------------------------
 
   void State::print(std::ostream & os) const {
     os << "insert diagnostic information about state here "
        << "(min/max/mean for each state var?)"
        << std::endl;
```

Comment:
output min/max/mean of the state variables to `os`. This will be used heavily in subsequent tests.

Reply:
also in regard to this, it's useful to print out the geometry before the state information... so ` os << *geom_ << std::endl << .... state_stuff....` or something like that

Reply:
@travissluka I will implement the print out at the next step.